src/sysdolphin/baselib/lobj.c

```diff
@@ -340,106 +285,1065 @@ void HSD_LObjSetup(HSD_LObj *lobj, GXColor color, f32 shininess, u32 unused)
+        lobj->shininess = 50.0F;
+        x = x = lobj->shininess;
+        x *= 0.5F;
+        GXInitLightAttn(&lobj->spec_lightobj, 0.0F, 0.0F, 1.0F, x, 0.0F, 1.0F - x);
+        switch (HSD_LObjGetType(lobj)) {
+        case 2:
+        case 3:
+            HSD_LObjGetPosition(lobj, &lobj->lvec);
+            PSMTXMUltiVec(mtx, &lobj->lvec, &lobj->lvec);
+            break;
+        case 1:
+            HSD_LObjGetLightVector(lobj, &lobj->lvec);
+            PSMTXMultVecSR(mtx, &lobj->lvec, &lobj->lvec);
+            PSVECNormalize(&lobj->lvec, &lobj->lvec);
+            break;
+        default:
+            __assert(lbl_804D5D18, 0x2A8, lbl_804D5D20);
+        }
+        lobj->flags |= 0x100;
+        lightmask_specular |= spec_id;
+    }
+}
+
+void setup_point_lightobj(HSD_LObj* lobj, Mtx mtx)
+{
+    Vec lpos;
+    GXInitLightColor(&lobj->lightobj, lobj->color);
+    lobj->hw_color = lobj->color;
+    HSD_LObjGetPosition(lobj, &lpos);
+    PSMTXMUltiVec(mtx, &lpos, &lpos);
+    GXInitLightPos(&lobj->lightobj, lpos.x, lpos.y, lpos.z);
+    GXInitLightPos(&lobj->spec_lightobj, lpos.x, lpos.y, lpos.z);
+    if (lobj->flags & 0x40) {
+        GXInitLightAttn(&lobj->lightobj, 1.0F, 0.0F, 0.0F,
+            lobj->u.attn.k0, lobj->u.attn.k1, lobj->u.attn.k2);
+    } else {
+        f32 ref_br = lobj->u.spot.ref_br;
+        f32 ref_dist = lobj->u.spot.ref_dist;
+        s32 dist_func = lobj->u.spot.dist_func;
+        GXInitLightDistAttn(&lobj->lightobj, ref_dist, ref_br, dist_func);
+        GXInitLightSpot(&lobj->lightobj, 0.0F, 0);
+        GXInitLightDistAttn(&lobj->spec_lightobj, ref_dist, ref_br, dist_func);
+    }
+}
+
+void setup_spot_lightobj(HSD_LObj* lobj, Mtx mtx)
+{
+    Vec lpos;
+    Vec ldir;
+    HSD_LObjGetPosition(lobj, &lpos);
+    PSMTXMUltiVec(mtx, &lpos, &lpos);
+    HSD_LObjGetLightVector(lobj, &ldir);
+    PSMTXMultVecSR(mtx, &ldir, &ldir);
+    PSVECNormalize(&ldir, &ldir);
+    GXInitLightPos(&lobj->lightobj, lpos.x, lpos.y, lpos.z);
+    GXInitLightPos(&lobj->spec_lightobj, lpos.x, lpos.y, lpos.z);
+    GXInitLightDir(&lobj->lightobj, ldir.x, ldir.y, ldir.z);
+    if (lobj->flags & 0x40) {
+        GXInitLightAttn(&lobj->lightobj,
+            lobj->u.attn.a0, lobj->u.attn.a1, lobj->u.attn.a2,
+            lobj->u.attn.k0, lobj->u.attn.k1, lobj->u.attn.k2);
+    } else {
+        f32 ref_br = lobj->u.point.ref_br;
+        f32 ref_dist = lobj->u.point.ref_dist;
+        f32 cutoff = lobj->u.point.cutoff;
+        s32 spot_func = lobj->u.spot.spot_func;
+        s32 dist_func = lobj->u.spot.dist_func;
+
+        GXInitLightDistAttn(&lobj->lightobj, ref_dist, ref_br, dist_func);
+        GXInitLightDistAttn(&lobj->spec_lightobj, ref_dist, ref_br, dist_func);
+        GXInitLightSpot(&lobj->lightobj, cutoff, spot_func);
+    }
+}
+
+extern char lbl_804D5D24[4];
 #pragma push
-asm void HSD_LObjSetup(HSD_LObj *lobj, GXColor color, f32 shininess, u32 unused)
-{ // clang-format off
+asm void HSD_LObjSetupInit()
+{
```

Comment:
Was this a mistake?

Reply:
Not sure what you mean?

Reply:
```diff
-asm void HSD_LObjSetup(HSD_LObj *lobj, GXColor color, f32 shininess, u32 unused)
-{ // clang-format off
+asm void HSD_LObjSetupInit()
+{
```

Reply:
Oh, the diff just shifted weirdly. Nevermind.

Reply:
HSD_LObjSetup is now decomped. HSD_LObjSetupInit is a different function.